src/main/java/green/citibike/mvc/CitiController.java

```diff
@@ -0,0 +1,88 @@
+package green.citibike.mvc;
+
+import green.citibike.aws.LambdaService;
+import green.citibike.aws.LambdaServiceFactory;
+import green.citibike.aws.Request;
+import green.citibike.aws.Response;
+import green.citibike.json.StationInfo;
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import org.jxmapviewer.JXMapViewer;
+import org.jxmapviewer.viewer.DefaultWaypoint;
+import org.jxmapviewer.viewer.GeoPosition;
+import org.jxmapviewer.viewer.Waypoint;
+import org.jxmapviewer.viewer.WaypointPainter;
+
+import javax.swing.*;
+import java.awt.geom.Point2D;
+import java.util.HashSet;
+import java.util.List;
+
+public class CitiController {
```

Comment:
You should construct this class with a lot of what it needs (mapViewer, textArea...) rather than passing it in each method call.